### ISSUE TITLE
Method of documenting organizations and individuals producing OSCAL Artifacts

### ISSUE THREAD

**Issue by david-waltermire-nist** [usnistgov/OSCAL#252]
# User Story:

As an OSCAL content creator, I have a means to document what organizations and/or individuals have a role in producing, reviewing, publishing, and maintaining an OSCAL artifact. As an OSCAL content consumer, I can identify what organizations and/or individuals had a role in producing, reviewing, publishing, and maintaining an OSCAL artifact.

## Goals:

1. Define a generalized mechanism for declaring organizations and individuals. This must include information sufficient to allow for contacting a referenced organization or individual (e.g., address, phone, email).
1. For all OSCAL artifacts, provide a means to associate specific organizations and individuals that have a role producing, reviewing, publishing, and maintaining the artifact.
1. For specific types of OSCAL artifacts (e.g., system implementations/SSPs, assessment results, etc) provide a means to associate specific organizations and individuals that have a role in that artifact. This will be done in the contents of the artifact specific model.
   - For system implementations roles include, but are not limited to: owning, operating, and authorizing the described system, Specific implementations may also need to define custom roles.
   - For assessments, the roles will need to identify who is performing the assessment for whom.

A rough example follows.
```
<artifact>
  <meta>
    <parties>
      <organization "NIST" party-id="NIST">
        <individual "Mark@NIST" party-id="mark@NIST"/>
        <individual "Joe@NIST" party-id="joe@NIST"/>
      </organization>
      <individual "Bob" party-id="BOB"/>
    </parties>

    <artifact-roles>
      <role name="prepared-by" party-ref="BOB">
    </artifact-roles>
  </meta>
  
  <system-roles> (SSP specific)
    <role id="SYSTEM-OWNER" name="system-owner" party-ref="mark@NIST"/>
    <role id="SYSTEM-OWNER" name="system-owner">
      <party party-ref="mark@NIST" status="primary"/>  
      <party party-ref="other@NIST" status="alternate"/>  
  </system-roles>

  <component>
    <component-roles>
      <role name="component-owner" role-ref="SYSTEM-OWNER"/>
      <role name="system-admin" party-ref="joe@NIST"/>
    </component-roles>
  </component>
</artifact>
```
## Dependencies:

None. Issue #196 contains some incomplete prior work related to this issue.

## Acceptance Criteria

1. The metamodels for the Catalog, Profile, and Implementation layers have been updated to incorporate this documentation mechanism.
1. All related model features are fully documented in each metamodel.
1. Examples of using this new feature are provided in each metamodel.

**Comment by bsilberberg** [comment]
I think this concept is much bigger then you think. I think it needs to be able to be applied to any object. This allows for track-ability of who the originator of the content was. You guys are doing 800-53. What if someone else makes their own security policies and forms to the same schema?

**Comment by wendellpiez** [comment]
**Sprint 15 Progress Oct 25 2018**

We are behind on this, waiting on improvements in metaschema infrastructure to support it.

These are nearing completion in branch https://github.com/wendellpiez/OSCAL/tree/feature-metaschema-d2-next. Metaschemas need to support a *manageable* modular architecture.

In the meantime, this should not be considered a blocker for other modeling initiatives; at this point permitting developers to improvise a little may actually be informative.

As soon as I have unblocked the metaschema I will make a branch to work with @brianrufgsa on this Issue.

Done: 20%

**Comment by david-waltermire-nist** [comment]
10/25/2018 Status
---
@david-waltermire-nist and @wendellpiez will meet to discuss this next week.

**Comment by iMichaela** [comment]
## 11/08/2018 ##
@wendellpiez worked on it and the WIP is available at: https://github.com/wendellpiez/OSCAL/tree/feature-metadata-schema/schema/metaschema for the team to review. @brianrufgsa's and @david-waltermire-nist's feedback is desired.

**Comment by david-waltermire-nist** [comment]
11/15/2018
---

PR #264 has been created to facilitate review, but work is not yet complete. @david-waltermire-nist and @brianrufgsa will review the work that has been completed. @wendellpiez will address any feedback before closing this issue.

Need to create issues for:
- Need more samples and testing of the model.
- Integrate the metadata schema into catalog, profile, and implementation schema.

**Comment by iMichaela** [comment]
## 11/27/2018 ##
The [WIP] PR #264 has only 10 files that have been modified and there is no <artifact> implementation in any of these files. I need to talk to @wendellpiez or @david-waltermire-nist .
Regarding the example above - we might need to support multiple roles for a system or subsystems of the overall system. Also, we might want to support the assessment planning (SAP) - and the assessment results (SAR)-related artifacts.

**Comment by wendellpiez** [comment]
@iMichaela Yes, these files represent modeling of *generalized file metadata* for any OSCAL resources, including catalogs, profiles and at higher layers such as implementation, SAP and SAR. As such it addresses requirements on this Issue but also #249 and #250.

Finalizing this will require not only (provisionally) validating the model itself, but also providing correct metadata to OSCAL artifacts including SP800-53 and FedRAMP examples.

The model in this PR does include a simple and very general mechanism for documenting and assigning roles, which could be used in the implementation layer. But only the metadata is provided for here, not the guts of an implementation description.

**Comment by akarmel** [comment]
**12/6/2018 - Sprint 16 Progress Notes**
- Per Dave/Wendell, recommend closing this issue as part of the merge of PR #264

**Comment by brianrufgsa** [comment]
12/20/2018 Wendell submitted PR #277, which requires an initial review from @david-waltermire-nist , and possible follow-up reviews before being fully accepted.

**Comment by wendellpiez** [comment]
Agree with @brianrufgsa that this is now stable; a new Issue can be made for new modeling requirements. (Issue is closable.)

**Comment by david-waltermire-nist** [comment]
Ok. Considering this closed.
